mteb/models/instruct_wrapper.py

```diff
@@ -78,3 +79,88 @@ def encode(
             return embeddings
 
     return InstructWrapper(model_name_or_path, mode, instruction_template, **kwargs)
+
+
+class InstructSentenceTransformerWrapper(Wrapper):
+    def __init__(
+        self,
+        model_name: str,
+        revision: str,
+        instruction_template: str | Callable[[str], str] | None = None,
+        max_seq_length: int | None = None,
+        apply_instruction_to_passages: bool = True,
+        padding_side: str | None = None,
+        add_eos_token: bool = False,
+        **kwargs: Any,
+    ):
+        """Instruct Sentence Transformer Wrapper. Wrapper that passes instructions to the Sentence Transformer model.
+        Applied for models like NV-Embed, gte-Qwen, e5-mistral, etc.
+
+        Arguments:
+            model_name: Model name
+            revision: Revision
+            instruction_template: Model template. Should contain the string '{instruction}'.
+            max_seq_length: Maximum sequence length. If None, the maximum sequence length won't be changed.
```

Comment:
Could we be more specific about _what_ it might be changed from, e.g.
```suggestion
            max_seq_length: Maximum sequence length. If None, the maximum sequence length will be read from the model config.
```